### PR TITLE
:hammer: refactoring + handle boolean prop-type

### PR DESCRIPTION
**Pull request by fabienjuif**
#41

### CODE REVIEW

**Comment by fabienjuif** [comment]
I let you read this version @bpetetot 
Other types will be added to (an) other(s) PR.

File: src/gui/components/props/input/input.container.js [hunk 1]
```diff
@@ -1,15 +1,44 @@
...
+
+const getValue = (type, e) => {
+  switch (type) {
+    case 'checkbox': return e.target.checked
+    default: return e.target.value
+  }
+}
+
+const mapState = ({ model }, { name }) => {
+  const m = model[name]
```
Comment (bpetetot) [inline src/gui/components/props/input/input.container.js]: `const { type, value } = model[name]`

File: src/gui/components/props/input/input.container.js [hunk 2]
```diff
@@ -1,15 +1,44 @@
...
+  }
+}
 
 const mapDispatch = (dispatch, { name }) => ({
-  onChange: event => dispatch({
-    type: 'SET_STATE',
-    payload: { [name]: event.target.value },
+  onChange: type => event => dispatch({
+    type: 'SET_VALUE',
+    payload: {
```
Comment (bpetetot) [inline src/gui/components/props/input/input.container.js]: same action in `src/gui/redux/model/model.actions.js` ?

File: src/gui/redux/model/model.actions.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const SET_MODEL = 'SET_MODEL'
+export const setModel = model => ({ type: SET_MODEL, payload: model })
+
+export const SET_VALUE = 'SET_VALUE'
+export const setValue = value => ({ type: SET_VALUE, payload: value })
```
Comment (bpetetot) [inline src/gui/redux/model/model.actions.js]: missing `name` property in the payload

**Review by bpetetot — DISMISSED** [review]
👍 only little changes, that can be done in next PRs if you want

**Review by bpetetot — APPROVED** [review]
👍

**Pull request merged**